cli/provider/cmd.go

```diff
@@ -456,15 +475,111 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 	c.logger.Debug("config has been initialised", zap.Any("for cmd", cmd.Name()), zap.Any("config", c.cfg))
 
 	switch cmd.Name() {
+	case "generate", "download":
+		path, err := cmd.Flags().GetString("path")
+		if err != nil {
+			errMsg := "failed to get the path"
+			utils.LogError(c.logger, err, errMsg)
+			return errors.New(errMsg)
+		}
+
+		if path == "" {
```

Comment:
Even if the user provides the path you should get the abs path. So you can just remove the if else condition and call the utils.ToAbsPath